### DISCUSSION TITLE
Debuggability of the mocks

### DISCUSSION THREAD

**Discussion by jsk95** [ashmind/SourceMock#11] [Q&A]
I think one of the benefits of the generated mocks (instead of in memory) is that users can easily debug the mocks if something isn't running as expected. Is this something the current design could support? This might not be easy currently as most of the storage & match is off loaded to internal classes

**Comment by ashmind** [comment]
Interesting question. I would probably prefer Source Link (symbol server) over moving more logic to generated code.
Generated code is quite hard to get right, so I would say the current design prefers to do as much as possible in inner code.
